third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -209,14 +212,20 @@ std::vector<ComputationClient::DataPtr> PjRtComputationClient::TransferToServer(
                 literal_pointer->shape().dimensions(), byte_strides,
                 xla::PjRtClient::HostBufferSemantics::
                     kImmutableUntilTransferCompletes,
-                [literal{std::move(literal)}]() { /* frees literal */ },
+                [literal{std::move(literal)}, &completed_transfers, &returned_future, total_tensors = tensors.size()]() { 
```

Comment:
Can we make this more simple by copying the `shared_ptr` to the `TimedSection` into each instance of this lambda? Then when the ref count hits 0 (ie all transfer futures have finished), the `shared_ptr` will destruct.

Reply:
Done!